src/de/error.rs

```diff
@@ -15,29 +16,71 @@ impl Unexpected for Number {
     }
 }
 
-impl Unexpected for Content<'_> {
-    fn unexpected(self, expected: Expected) -> Error {
-        let found = match self {
+impl Content<'_> {
+    fn into_found(self) -> Found {
+        match self {
             Content::Unit => Found::Unit,
             Content::Bool(v) => Found::Bool(v),
             Content::Number(v) => Found::Number(v),
             Content::Char(v) => Found::Char(v),
             Content::String(v) => Found::String(v.into_owned()),
             Content::Bytes(v) => Found::Bytes(v.into_owned()),
-            Content::Seq(_) => Found::Seq,
-            Content::Map(_) => Found::Map,
-            Content::Option(_) => Found::Option,
+            Content::Seq(v) => {
+                let mut vec = Vec::with_capacity(v.len());
+                for content in v {
+                    vec.push(content.into_found());
+                }
+                Found::Seq(vec)
+            }
+            Content::Map(v) => {
+                let mut vec = Vec::with_capacity(v.len());
+                for (key, value) in v {
+                    vec.push((key.into_found(), value.into_found()));
+                }
+                Found::Map(vec)
+            }
+            Content::Option(v) => Found::Option(v.map(|x| Box::new(x.into_found()))),
             Content::Struct(v) => Found::Struct {
                 name: v.name.to_owned(),
-                typ: v.data.typ(),
+                data: Box::new(v.data.into_found()),
```

Comment:
Is there a possible impact in performances due to the use of heap here?

Reply:
The heap allocation is necessary because the type would be recursive otherwise. The type of `data` is defined as

```rust
pub enum Data {
    Unit,
    NewType(Found),
    Tuple(Vec<Found>),
    Struct(Vec<(String, Found)>),
}
```

As you can see, this references `Found` so we need a box to break that recursion.